consensus_encoding/src/encode/encoders.rs

```diff
@@ -59,6 +60,11 @@ impl<const N: usize> Encoder for ArrayEncoder<N> {
     }
 }
 
+impl<const N: usize> ExactSizeEncoder for ArrayEncoder<N> {
+    #[inline]
+    fn len(&self) -> usize { N }
```

Comment:
In 5fbf0a2fcfb4b9abcd87d30767e2e014726bcbb2:

I think we want to do `self.arr.map_or(0, |a| a.len())` here like we do with `CompactSizeEncoder`. The idea is that this should yield the exact length of the *remaining* bytes, since that's what's relevant if you're trying to hex encode.